engine/cdn/storage/redis/redis.go

```diff
@@ -40,13 +40,30 @@ func (s *Redis) ItemExists(i index.Item) (bool, error) {
 }
 
 func (s *Redis) Add(i index.Item, index uint, value string) error {
+	value = fmt.Sprintf("%d#%s", index, value)
 	return s.store.ScoredSetAdd(context.Background(), i.ID, value, float64(index))
 }
 
+func (s *Redis) Append(i index.Item, value string) error {
+	return s.store.ScoredAppend(context.Background(), i.ID, value)
+}
+
+func (s *Redis) Card(i index.Item) (int, error) {
+	return s.store.SetCard(i.ID)
+}
+
 func (s *Redis) Get(i index.Item, from, to uint) ([]string, error) {
 	var res = make([]string, to-from+1)
-	err := s.store.ScoredSetScan(context.Background(), i.ID, float64(from), float64(to), &res)
-	return res, err
+	if err := s.store.ScoredSetScan(context.Background(), i.ID, float64(from), float64(to), &res); err != nil {
+		return res, err
+	}
+	for i := range res {
+		linesSplitted := strings.SplitAfterN(res[i], "#", 2)
```

Comment:
I wonder is this is more efficient:
res[i] = strings.TrimLeftFunc(res[i], func(r rune) bool {
     return unicode.IsDigit(r) || unicode.IsNumber(r) || r == rune('#')
})

Reply:
Will use:  
res[i] = strings.TrimFunc(res[i], unicode.IsNumber)
res[i] = strings.TrimPrefix(res[i], "#")